test/integration/targets/github_issue/tasks/main.yml

```diff
@@ -1,81 +1,73 @@
-# it under the terms of the GNU General Public License as published by
-# the Free Software Foundation, either version 3 of the License, or
-# (at your option) any later version.
-#
-# Ansible is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
 #
+# Copyright: (c) 2017-2018, Abhijeet Kasurde <akasurde@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- debug:
+    msg: "{{ ansible_os_family }} {{ ansible_distribution_major_version }}"
+
+- block:
+  - name: make sure github3.py is not installed
+    pip:
+      name: github3.py
+      state: absent
 
-- name: make sure github3.py is not installed
-  pip:
-    name: github3.py
-    state: absent
+  # Testcase 0001: Check if dependency is installed
+  - name: Check if github3.py is installed or not
+    github_issue:
+      organization: "{{ organization }}"
+      repo: "{{ repo }}"
+      issue: "{{ non_existent_issue }}"
+      action: get_status
+    register: lib_missing
+    ignore_errors: True
 
-# Testcase 0001: Check if dependency is installed
-- name: Check if github3.py is installed or not
-  github_issue:
-    organization: "{{ organization }}"
-    repo: "{{ repo }}"
-    issue: "{{ non_existent_issue }}"
-    action: get_status
-  register: lib_missing
-  ignore_errors: True
+  - assert:
+      that:
+        - "{{ lib_missing.failed == True }}"
+        - "{{ lib_missing.changed == False }}"
+        - "{{ 'Missing required github3 module.' in lib_missing.msg }}"
 
-- assert:
-    that:
-      - "{{ lib_missing.failed == True }}"
-      - "{{ lib_missing.changed == False }}"
-      - "{{ 'Missing required github3 module.' in lib_missing.msg }}"
+  - include: '{{ item }}'
+    with_first_found:
+      - files:
+          - '{{ ansible_distribution }}-{{ ansible_distribution_version }}-python{{ ansible_python_version.split(".")[0] }}.yml'
+          - '{{ ansible_distribution }}-python{{ ansible_python_version.split(".")[0] }}.yml'
+          - '{{ ansible_os_family }}-python{{ ansible_python_version.split(".")[0] }}.yml'
+          - 'default-python{{ ansible_python_version.split(".")[0] }}.yml'
 
-- include: '{{ item }}'
-  with_first_found:
-    - files:
-        - '{{ ansible_distribution }}-{{ ansible_distribution_version }}-python{{ ansible_python_version.split(".")[0] }}.yml'
-        - '{{ ansible_distribution }}-python{{ ansible_python_version.split(".")[0] }}.yml'
-        - '{{ ansible_os_family }}-python{{ ansible_python_version.split(".")[0] }}.yml'
-        - 'default-python{{ ansible_python_version.split(".")[0] }}.yml'
+  - name: make sure github3.py is installed
+    pip:
+      name: github3.py
```

Comment:
Include a test against a version of github3.py < 1.0.0 to cover the new version checking code path.

Do this after the test with github3.py removed, but before the latest version is installed.

Reply:
github3.py < 1.0.0 are not stable one. So I think we should stick to current stable version and support it.

Reply:
I'm OK with not supporting github3.py < 1.0.0 -- but there is code in the module to check for that, so we should at least have a test to verify the error is handled correctly.